lemur_vault/plugin.py

```diff
@@ -300,6 +302,69 @@ def authenticate_certificate():
         raise Exception('Vault Config: cert or key path not set.')
 
 
+def generate_gcp_jwt():
+    """
+    Generate GCP JWT for Vault authentication.
+    :return: GCP JWT
+    """
+    role = current_app.config.get('VAULT_AUTH_ROLE')
+    account = current_app.config.get('VAULT_AUTH_ACCOUNT')
+
+    if role and account:
+        headers = {'Metadata-Flavor': 'Google'}
+        url = 'http://metadata/computeMetadata/v1/instance/service-accounts/' + account + '/identity'
+        try:
+            data = [('audience', current_app.config.get('VAULT_URL') + '/vault/' + role ),('format', 'full')]
+            resp = requests.post(url, headers=headers, data=data)
+            
+            if resp.status_code != 200 and resp.status_code != 204:
+                current_app.logger.info('Vault: ' + resp.text)
+                raise Exception('Vault GCP Auth: Issues retrieving JWT.')
+
+            return resp.text
+
+        except ConnectionError as ConnError:
+            current_app.logger.info('Vault: There was an error while connecting to GCE metadata.')
+            raise ConnError
+
+    else:
+        raise Exception('Vault Config: Role and Service Account not set.')
+
+
+
+def authenticate_gcp():
+    """
+    GCP JWT authentication function.
+    :return: Client token.
+    """
+    role = current_app.config.get('VAULT_AUTH_ROLE')
+
+    if role:
+        url = current_app.config.get('VAULT_URL') + '/v1/auth/gcp/login'
```

Comment:
This assumes that the GCP auth backend in Vault is mounted at the default path of `gcp`. It would be nice if an alternate mount point could optionally specified. See the below code in the hvac library for an example of how to do this.

https://github.com/hvac/hvac/blob/master/hvac/v1/__init__.py#L1199-L1221

Reply:
@ecterun please add mounting point parameters, it's part of the authentication requirements from now on.
New issues will be opened on the old authentication methods